### ISSUE TITLE
[REGRESSION]: VSTest@3 Task version 3.245.1 is not finding any test cases

### ISSUE THREAD

**Issue by rygo-msft** [microsoft/azure-pipelines-tasks#20390]
### New issue checklist

- [X] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [X] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [X] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

VSTest@3

### Breaking task version

3.245.1

### Last working task version

3.231.0

### Regression Description

The default source search filter was changed in a non-major task version update. This has broken existing pipelines that are now finding the same set of tests as they were before. Please reconsider this change.
 
Test Source Filter in 3.231.0
Source filter: **\*test*.dll,!**\*TestAdapter.dll,!**\obj\**

New test source filter in 3.245.1
Source filter: **\bin\**\*test.dll,**\bin\**\*tests.dll

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [ ] Microsoft Hosted
- [X] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Windows Server 2022

### Relevant log output

```shell
Starting: VSTest
==============================================================================
Task         : Visual Studio Test
Description  : Run unit and functional tests (Selenium, Appium, Coded UI test, etc.) using the Visual Studio Test (VsTest) runner. Test frameworks that have a Visual Studio test adapter such as MsTest, xUnit, NUnit, Chutzpah (for JavaScript tests using QUnit, Mocha and Jasmine), etc. can be run. Tests can be distributed on multiple agents using this task (version 2 and later).
Version      : 3.245.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/test/vstest
==============================================================================
Running tests using vstest.console.exe runner.
======================================================
Test selector : Test assemblies
Test filter criteria : ************************
Search folder : *************************
Action when minimum tests threshold not met : fail
Minimum tests expected to be run: 1
VisualStudio version selected for test execution : latest
Attempting to find vstest.console from a visual studio installation with version [17.0,18.0).
Run in parallel : false
Run in isolation : false
Path to custom adapters : undefined
Other console options : undefined
Code coverage enabled : false
Diagnostics enabled : false
Run the tests locally using vstest.console.exe
========================================================
Source filter: **\bin\**\*test.dll,**\bin\**\*tests.dll
##[warning]No test sources found matching the given filter '**\bin\**\*test.dll,**\bin\**\*tests.dll'
======================================================
##[error]VsTest task failed.
Finishing: VSTest
```

### Full task logs with system.debug enabled

<details>
  <summary>UNSUCCESSFUL RUN</summary>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

<details>
  <summary>SUCCESSFUL RUN</summary>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Repro steps

```yml
Run VSTest against dlls with names of the format *Tests.dll in a folder other than bin.
```

**Comment by hughbidone** [comment]
In the meantime, is there away to specify the custom (working) version in Classic pipeline? As this bug just completely renders the `3.*` version on Classic pipeline. Thanks.
